daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/PrimitivesElementKinds.scala

```diff
@@ -81,7 +82,14 @@ case class DelimiterStackCombinatorElement(e: ElementBase, body: Gram) extends T
   lazy val uInit = if (e.initiatorParseEv.isKnownNonEmpty) One(e.initiatorUnparseEv) else Nope
   lazy val uTerm = if (e.terminatorParseEv.isKnownNonEmpty) One(e.terminatorUnparseEv) else Nope
 
-  lazy val parser: DaffodilParser = new DelimiterStackParser((pInit.toList ++ pTerm.toList).toArray, e.termRuntimeData, body.parser)
+  lazy val delims = (pInit.toList ++ pTerm.toList)
+
+  override def toString() =
+    "<" + Misc.getNameFromClass(this) + " delims='" + delims.mkString(" ") + "'>" +
```

Comment:
In the past we've had issues where delimters aren't correctly escaped according to HTML standards. This makes running things through xmllint to see a pretty indented version difficult. Shoud/can we run the delims through some sort of HTML-ifier so they are more friendly to such tools?

Reply:
Yes, should escape these. 
scala.xml.Utility.escape is insufficient however, as that (a) doesn't escape apostrophes because HTML doesn't have an &apos; entity (b) delims might be XML-illegal characters.

So we have to remap all XML-illegal chars to XML-legal PUA chars, but leave CR and CRLF alone, remap C0 and C1 control characters appropriately to numeric entity refs. Remap whitespace chars to visible glyphs.

We have all the pieces of this algorithm, just not in this particular combination.

Reply:
Fixed in most recent commit.